app/code/Magento/Quote/Model/QuoteManagement.php

```diff
@@ -665,6 +665,10 @@ protected function _prepareCustomerQuote($quote)
         if ($shipping && !$shipping->getCustomerId() && !$hasDefaultBilling) {
             $shipping->setIsDefaultBilling(true);
         }
+
+        if (!$billing->getCustomerAddressId() && $shipping) {
```

Comment:
Could you change sequence of the condition?
`($shipping && !$billing->getCustomerAddressId())`

Thanks